src/components/app/app.tsx

```diff
@@ -1,22 +1,25 @@
-import {useIsConnectionRestored} from '@tonconnect/ui-react';
+import {useIsConnectionRestored, useTonAddress} from '@tonconnect/ui-react';
 import {useEffect} from 'react';
 
 import styles from './app.module.css';
 import {Home} from '../../screens/home/home.tsx';
 import {useDispatch} from '../../store';
 import {loadAssetsActions} from '../../store/assets/assets-actions.ts';
+import {balancesActions} from '../../store/balances/balances-actions.ts';
 import {Header} from '../header/header.tsx';
 
 const tg = window.Telegram.WebApp;
 
 export const App = () => {
     const dispatch = useDispatch();
     const connectionRestored = useIsConnectionRestored();
+    const walletAddress = useTonAddress();
 
     useEffect(() => {
         tg.ready();
         dispatch(loadAssetsActions.submit());
-    }, [dispatch]);
+        dispatch(balancesActions.submit({walletAddress}));
+    }, [dispatch, walletAddress]);
```

Comment:
you will re-call tg.ready()and re-dispatch assets loading actions  on each wallet connection